cumulus_library/study_parser.py

```diff
@@ -219,49 +259,25 @@ def run_table_builder(
         :param schema: The name of the schema to write tables to
         :param verbose: toggle from progress bar to query output
         """
-
-        # Since we have to support arbitrary user-defined python files here, we
-        # jump through some importlib hoops to import the module directly from
-        # a source file defined in the manifest.
-
-        # As with eating an ortolan, you may wish to cover your head with a cloth.
-        # Per an article on the subject: "Tradition dictates that this is to shield
-        # – from God’s eyes – the shame of such a decadent and disgraceful act."
         for file in self.get_table_builder_file_list():
-            # Grab the BaseTableBuilder class from the manifest-defined module
-            spec = importlib.util.spec_from_file_location(
-                "table_builder", f"{self._study_path}/{file}"
-            )
-            table_builder_module = importlib.util.module_from_spec(spec)
-            sys.modules["table_builder"] = table_builder_module
-            spec.loader.exec_module(table_builder_module)
-
-            # We're going to find all subclasses of BaseTableBuild in this file.
-            # Since BaseTableBuilder itself is a valid subclass of BaseTableBuilder,
-            # we'll detect and skip it. If we don't find exactly one subclass,
-            # we'll punt.
-            table_builder_subclasses = []
-            for _, cls_obj in inspect.getmembers(table_builder_module, inspect.isclass):
-                if (
-                    issubclass(cls_obj, BaseTableBuilder)
-                    and cls_obj != BaseTableBuilder
-                ):
-                    table_builder_subclasses.append(cls_obj)
-            if len(table_builder_subclasses) != 1:
-                raise StudyManifestParsingError(
-                    f"Error loading {self._study_path}{file}\n"
-                    "Custom builders must extend the BaseTableBuilder class exactly once per module."
-                )
-
-            # We'll get the subclass, initialize it, run the executor code, and then
+            table_builder, table_builder_module = self.load_builder_from_file(file)
+            # After runnning the executor code, we'll remove the builder
             # remove it so it doesn't interfere with the next python module to
             # execute, since the subclass would otherwise hang around.
-            table_builder_class = table_builder_subclasses[0]
-            table_builder = table_builder_class()
+
             table_builder.execute_queries(cursor, schema, verbose)
             del sys.modules[table_builder_module.__name__]
             del table_builder_module
```

Comment:
This cleanup feels like it belongs to the moved function - and you'd want this same cleanup for `run_single_table_builder` in theory yeah? (if the program didn't just exit after - e.g unit tests at least). Is there a clean way to group this with the moved logic?

Maybe a contextlib.contextmanager would let you handle cleanup cleanly - so like:

```
with load_builder() as builder:
    table_builder.execute_queries(cursor, schema, verbose)
```

Or just combine the loading and executing together anyway? Is the only difference between these two methods whether `drop_table` gets passed?